src/GraphQL/Validation/DocumentValidator.ParseArgumentVisitor.cs

```diff
@@ -0,0 +1,135 @@
+using GraphQL.Execution;
+using GraphQL.Types;
+using GraphQLParser.AST;
+using GraphQLParser.Visitors;
+
+namespace GraphQL.Validation;
+
+public partial class DocumentValidator
+{
+    private class ParseArgumentVisitor : ASTVisitor<ParseArgumentVisitor.Context>
+    {
+        public static readonly ParseArgumentVisitor Instance = new();
+
+        private ParseArgumentVisitor() { }
+
+        protected override async ValueTask VisitOperationDefinitionAsync(GraphQLOperationDefinition operationDefinition, Context context)
```

Comment:
I have a suspicion that this code should be called only for executed operation - that is returned from `DocumentExecuter.GetOperation`.

Reply:
It will save cpu cycles for sure, but nor does it hurt.  While easy to do here, it is not quite so easy to selectively parse fragments that are in use.  I can write such logic, however.  Should I?

Reply:
Done

Reply:
It's the next on my todo/review list.

Reply:
It's the next on my todo/review list.